app/services/transfers/service.go

```diff
@@ -126,106 +128,133 @@ func (bs *Service) SaveRecoveredTxn(txId, amount string, asset string, m memo.Me
 		Asset:         asset,
 	})
 	if err != nil {
-		bs.logger.Errorf("Something went wrong while saving new Recovered Transaction with ID [%s]. Err: [%s]", txId, err)
+		ts.logger.Errorf("Something went wrong while saving new Recovered Transaction with ID [%s]. Err: [%s]", txId, err)
 		return err
 	}
 
-	bs.logger.Infof("Added new Transaction Record with Txn ID [%s]", txId)
+	ts.logger.Infof("Added new Transaction Record with Txn ID [%s]", txId)
 	return err
 }
 
 // VerifyFee verifies that the provided TX reimbursement fee is enough using the
 // Fee Calculator and updates the Transaction Record to Insufficient Fee if necessary
-func (bs *Service) VerifyFee(tm encoding.TransferMessage) error {
-	isSufficient, err := bs.fees.ValidateExecutionFee(tm.Fee, tm.Amount, tm.GasPriceGwei)
+func (ts *Service) VerifyFee(tm encoding.TransferMessage) error {
+	isSufficient, err := ts.fees.ValidateExecutionFee(tm.Fee, tm.Amount, tm.GasPriceGwei)
 	if !isSufficient {
-		bs.logger.Errorf("Fee validation for TX ID [%s] failed. Provided tx reimbursement fee is invalid/insufficient. Error [%s]", tm.TransactionId, err)
-		err = bs.transactionRepository.UpdateStatusInsufficientFee(tm.TransactionId)
+		ts.logger.Errorf("Fee validation for TX ID [%s] failed. Provided tx reimbursement fee is invalid/insufficient. Error [%s]", tm.TransactionId, err)
+		err = ts.transactionRepository.UpdateStatusInsufficientFee(tm.TransactionId)
 		if err != nil {
-			bs.logger.Errorf("Failed to update status to [%s] of transaction with TransactionID [%s]. Error [%s].", transaction.StatusInsufficientFee, tm.TransactionId, err)
+			ts.logger.Errorf("Failed to update status to [%s] of transaction with TransactionID [%s]. Error [%s].", transaction.StatusInsufficientFee, tm.TransactionId, err)
 			return err
 		}
-		bs.logger.Debugf("TX with ID [%s] was updated to [%s]. Provided fee [%s]", tm.TransactionId, transaction.StatusInsufficientFee, tm.Fee)
+		ts.logger.Debugf("TX with ID [%s] was updated to [%s]. Provided fee [%s]", tm.TransactionId, transaction.StatusInsufficientFee, tm.Fee)
 		return err
 	}
 	return nil
 }
 
-// ProcessTransfer processes the transfer message by signing the required
-// authorisation signature submitting it into the required HCS Topic
-func (bs *Service) ProcessTransfer(tm encoding.TransferMessage) error {
-	authMsgHash, err := auth_message.EncodeBytesFrom(tm.TransactionId, tm.EthAddress, tm.Erc20Address, tm.Amount, tm.Fee)
+func (ts *Service) submitAuthorizationMessage(tm encoding.TopicMessage) error {
```

Comment:
It seems strange that `submitAuthorizationMessage` does:
- submit
- getting tx from db
- updating tx record
- creating callbacks
- Attaching callbacks to transaction for monitoring

I don't have a strong opinion about it, but to me it seems that we might be better off if the logic in this function is placed in `ProcessTransfer`. Not to mention that the error logs are redundant since we have a 2 layer logic here (1st layer `ProcessTransfer`, 2nd layer `submitAuthorizationMessage`